### PR TITLE
Fix/expose segments

### PR DESCRIPTION
**Pull request by jamieforth**
This addresses a couple of small issues with segment metadata.

When `dejitter_timestamps=False` reported segments now have the correct sample stop. 

When `dejitter_timestamps=True`
- non-empty streams no longer contain duplicate segment tuples
- empty stream now return segments as an empty array, rather than [0, -1]

### CODE REVIEW

**Comment by cbrnr** [comment]
@agricolab do you have time to look?

File: pyxdf/pyxdf.py [hunk 1]
```diff
@@ -629,6 +629,7 @@ def _jitter_removal(streams, threshold_seconds=1, threshold_samples=500):
     for stream_id, stream in streams.items():
         stream.effective_srate = 0  # will be recalculated if possible
         nsamples = len(stream.time_stamps)
+        stream.segments = []
```
Comment (agricolab) [inline pyxdf/pyxdf.py:632]: 👌🏼 catches empty streams and returns [] instead of [0,-1] (when `_jitter_removal` was called, i.e. `dejitter_timestamps=True`

File: pyxdf/pyxdf.py [hunk 2]
```diff
@@ -348,7 +348,7 @@ def load_xdf(
             if stream.fmt == "string":
                 stream.time_series = []
             else:
-                stream.time_series = np.zeros((stream.nchns, 0))
+                stream.time_series = np.zeros((0, stream.nchns))
```
Comment (agricolab) [inline pyxdf/pyxdf.py:351]: 👌🏼  good catch. we probably need a test for empty streams eventually to catch such issues

**Comment by agricolab** [comment]
Did a manual test and it passes now unduplicated segments

```
import pyxdf; 
s, h = pyxdf.load_xdf("minimal.xdf", jitter_break_threshold_seconds=0.001, jitter_break_threshold_samples=1);
assert s[0]["info"]["segments"] == [(0, 0), (1, 3), (4, 8)]
```

**Comment by agricolab** [comment]
Thanks @jamieforth !

**Comment by cbrnr** [comment]
Thanks @jamieforth and @agricolab! Does the test you mention pass with this change, but fail previously? If so, could you add it to `pyxdf/test/test_data.py`?

**Comment by agricolab** [comment]
Yes, could be a good regression test. I'd also like if we'd had a minimal.xdf (or empty.xdf) with an empty stream.
